config/src/network_id.rs

```diff
@@ -152,6 +152,64 @@ impl NetworkId {
         matches!(self, NetworkId::Validator)
     }
 
+    pub fn upstream_roles(&self, role: &RoleType) -> &'static [PeerRole] {
+        match self {
+            NetworkId::Validator => match role {
+                RoleType::Validator => &[PeerRole::Validator],
+                RoleType::FullNode => &[],
+            },
+            NetworkId::Public => &[
+                PeerRole::PreferredUpstream,
+                PeerRole::Upstream,
+                PeerRole::ValidatorFullNode,
+            ],
+            NetworkId::Private(_) => {
+                if self.is_vfn_network() {
+                    match role {
+                        RoleType::Validator => &[],
+                        RoleType::FullNode => &[PeerRole::Validator, PeerRole::ValidatorFullNode],
+                    }
+                } else {
+                    // TODO: This is not a well defined network
+                    &[]
+                }
```

Comment:
seems like this should be the same as Public? or at least PreferredUpstream and then Upstream, no?

Reply:
Yeah, I was debating.  Probably just make a `Default` set that applies to this and public.